lms/djangoapps/discussion_api/api.py

```diff
@@ -391,7 +391,10 @@ def get_comment_list(request, thread_id, endorsed, page, page_size, mark_as_read
     # existing comments service interface
     if cc_thread["thread_type"] == "question":
         if endorsed is None:
-            raise ValidationError({"endorsed": ["This field is required for question threads."]})
+            endorsed_responses = cc_thread["endorsed_responses"][response_skip:(response_skip + page_size)]
+            non_endorsed_responses = cc_thread["non_endorsed_responses"]
+            responses = endorsed_responses + non_endorsed_responses
+            resp_total = len(endorsed_responses) + len(non_endorsed_responses)
```

Comment:
This won't work properly with pagination.  
The non-endorsed-responses will be returned with each page of the endorsed response.
Did you test with there being multiple pages of endorsed responses?
The expected behavior would be:
endorsed-response Page 1
endorsed-response Page 2
...
endorsed-response Page N
non-endorsed-response Page N + 1
...
non-endorsed-response Page N + M